### ISSUE TITLE
All your hints are belong to us

### ISSUE THREAD

**Issue by rsheeter** [googlefonts/fontmake-rs#51]
In https://github.com/googlefonts/fontmake-rs/issues/33#issuecomment-1346920324 concerns about hints came up. It then became unclear what the existing fontmake tooling actually supports. Based on #33 and IM conversation I current believe that:

* Glyphs can embed hinting information but fontmake wouldn't read or compile it
* Tools exist to stick hints directly into the binary (or ttx), any outline edit would presumably invalidate the hints
   * @davelab6 noted "the hint byte code is turned into assembly in .TTX files, and can be read in a text editor; and fontforge provides a step debugger for TT assembly via FreeType."
*  People aspire to stick hint information into UFOs but fontmake might (*TODO clarify*) not read or compile it
   * See https://unifiedfontobject.org/versions/ufo3/lib.plist/#publictruetypeinstructions, https://unifiedfontobject.org/versions/ufo3/glyphs/glif/#publictruetypeinstructions. 
   * https://github.com/googlefonts/ufo2ft/pull/654 aims to add fontmake support for ^

(will update as I learn more)

**Comment by anthrotype** [comment]
> Glyphs can embed hinting information but fontmake wouldn't read or compile it

that's right, fontmake can _not_ yet read nor compile [Glyphs.app](http://glyphs.app/) hinting, as glyhpsLib even crashes on them: https://github.com/googlefonts/glyphsLib/issues/817

> People aspire to stick hint information into UFOs but fontmake might not read or compile it

correct, until ttps://github.com/googlefonts/ufo2ft/pull/654

**Comment by rsheeter** [comment]
Additional context courtesy of Adam:

> The FontLab compiler from visual commands to binary is proprietary to FontLab. This works well for environments without subpixel rendering & VF.
> 
> The Glyphs compiler from visual commands to binary is proprietary to Glyphs. It's relatively lightweight set of visual commands that I think work with VF. 
> 
> The VTT compiler was opemsourced by MS not so long ago, and that's the first (and good) piece of code that does the compilation.
> 
> But there isn't really any established way to store VTT visual commands in source formats (should be)
> 
> And there isn't any code to convert from the Glyphs visual commands or from the FontLab visual commands to VTT commands, but since both G & FL commands are kind of a subset of VTT, such converters could be made

Marc adds

> gftools builder uses vttLib. We don't store the hinting info in the ufos since we also build glyphs files. Instead, the user provides a mapping which maps fonts to ttx files e.g https://github.com/wix/wixmadefor/blob/main/sources/config_text.yaml#L20-L36
